frontends/web/src/components/headerssync/headerssync.tsx

```diff
@@ -14,44 +16,61 @@
  * limitations under the License.
  */
 
-import { Component, h } from 'preact';
-import { translate } from 'react-i18next';
+import { Component, h, RenderableProps } from 'preact';
 import { subscribe } from '../../decorators/subscribe';
+import { translate, TranslateProps } from '../../decorators/translate';
+import { CoinCode } from '../../routes/account/account';
+import Spinner from '../spinner/ascii';
 import * as style from './headerssync.css';
 
-@translate()
-// @ts-ignore (generics need to be typed explicitly once converted to TypeScript)
-@subscribe(props => ({ status: 'coins/' + props.coinCode + '/headers/status' }))
-export default class HeadersSync extends Component {
-    constructor(props) {
-        super(props);
-        this.state = { show: 0 };
+interface HeadersSyncProps {
+    coinCode: CoinCode;
+}
+
+interface StatusInterface {
+    targetHeight: number;
+    tip: number;
+    tipAtInitTime: number;
+    tipHashHex: string;
+}
+
+interface SubscribedHeadersSyncProps {
+    status?: StatusInterface;
+}
+
+interface State {
+    show: number;
+}
+
+type Props = SubscribedHeadersSyncProps & HeadersSyncProps & TranslateProps;
+
+class HeadersSync extends Component<Props, State> {
+    public readonly state: State = {
+        show: 0,
     }
 
     componentDidUpdate(prevProps) {
-        const status = this.props.status;
+        const { status } = this.props;
         if (status && prevProps.status && status.tip !== prevProps.status.tip) {
-            this.setState({ show: status.tip }); // eslint-disable-line
+            this.setState({ show: status.tip });
             if (status.tip === status.targetHeight) {
-                setTimeout(() => this.setState(state => state.show === status.tip && { show: 0 }), 4000);
+                setTimeout(() => this.setState(state => state.show === status.tip ? { show: 0 } : null), 4000);
```

Comment:
Since w'ere here, what's this magic 4000ms? I mean, why 4000?

Reply:
commented `hide component after 4s when tip reached targetHeight`, but no idea why 4s was chosen and not 5s.

It used to be 5s :) https://github.com/digitalbitbox/bitbox-wallet-app/commit/499ad0f2ed6d956c261e63b2abee9ac6b2fc8456#diff-11692a3eba9e705270b5c20ffbad8be0ca2b85a4e380a0f3a945de78c442859b